### DISCUSSION TITLE
Libre 2 "NFC read timeout"

### DISCUSSION THREAD

**Discussion by Etudes-NG** [NightscoutFoundation/xDrip#2839] [General]
Hello, 

Xdrip update yesterday and today when starting a new sensor this information appears briefly "NFC read timeout".
I follow this instructions when starting a new sensor: https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip?
Sensor work properly with freestyle app.

Any help appreciated. 

Nathanael

**Comment by Etudes-NG** [comment]
Tested using previous xdrip release, same error. Maybe the problem is coming from a the sensor as stated in https://github.com/NightscoutFoundation/xDrip/discussions/2840

**Reply by PhilTU** [reply]
Error / event log would help. That would say if it was unknown sensor type issue

**Reply by aquinohugo** [reply]
Hello,

I'm having the exact same issue. Today I changed the libre2 sensor and now it won't read the nfc on the xdrip app. Only on the freestyle app. It can't be a problem with the sensor.

**Reply by aquinohugo** [reply]
![Screenshot_2023-05-29-02-31-30-662_com whatsapp](https://github.com/NightscoutFoundation/xDrip/assets/73556027/ce6d8744-ac09-4d71-bb5a-7a24379932fd)
On the event log, it says "sensor type unknown" and returners it as a libre1, but in fact it is a libre2

**Reply by Navid200** [reply]
Wasn't this why we had this release?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2023.05.25

**Reply by PhilTU** [reply]
yes, the error is the unknown sensor type defaulting to Libre 1 when it's actually a Libre 2. @aquinohugo can you update to latest nightly build and retest.

**Reply by aquinohugo** [reply]
Just installed the latest version (26 May build) and now it can detect the sensor as a libre2. However, it still gives the NFC timeout error and I am unable to start the sensor. Any idea? 
![IMG-20230529-WA0004](https://github.com/NightscoutFoundation/xDrip/assets/73556027/e50a2b61-6e1c-4e15-853b-3f6df1c87036)

**Reply by PhilTU** [reply]
Did you also update OOP2 as per the linked discussion?

**Reply by aquinohugo** [reply]
IT FINALY WORKED!

I thought I was using the latest version from oop2. I was using the "[oop2.apk](https://drive.google.com/file/d/17BlRT9aL58hikOIr-MsVs57Lr8seuu5f/view?usp=sharing) - Version 3: 8a0ee8f-2022.06.10 (latest version)" from the guide introduced above by @Etudes-NG. However it wasn´t working.

It only start working when I used the OOP2 2023.05.25 version from @tzachi-dar comment (https://github.com/NightscoutFoundation/xDrip/discussions/2845#discussioncomment-5995246) in combination with the xDrip May 26 nightly build.

**Reply by Navid200** [reply]
@psonnera 
Just for your reference.  
I have no idea if you may like to update your guide to have Tzachi's OOP2, assuming you haven't already.

**Reply by psonnera** [reply]
@Navid200 done the evening it was released 😃
